src/basic/terminal-util.c

```diff
@@ -1283,6 +1283,31 @@ int vt_reset_keyboard(int fd) {
         return 0;
 }
 
+int vt_restore(int fd) {
+        static const struct vt_mode mode = {
+                .mode = VT_AUTO,
+        };
+        int r;
+
+        r = ioctl(fd, KDSETMODE, KD_TEXT);
+        if (r < 0)
+                return -errno;
+
+        r = vt_reset_keyboard(fd);
+        if (r < 0)
+                return r;
+
+        r = ioctl(fd, VT_SETMODE, &mode);
+        if (r < 0)
+                return -errno;
+
+        r = fchown(fd, 0, (gid_t) -1);
+        if (r < 0)
+                return -errno;
```

Comment:
this code is quite different from the old: if any of the calls above fails we'll not bother with the rest of the calls. Previously, we'd continue with the next one... I am pretty sure we should continue to do that...

I think it would be a good idea to add log_debug_errno( )calls for each failing call, to make this easier to debug.

Reply:
OK